frontend/src/lib/utils.tsx

```diff
@@ -1283,21 +1338,28 @@ export function convertPropertyGroupToProperties(
     return properties
 }
 
-export function flattenPropertyGroup(
-    flattenedProperties: AnyPropertyFilter[],
-    propertyGroup: PropertyGroupFilter | AnyPropertyFilter
-): AnyPropertyFilter[] {
-    const obj: AnyPropertyFilter = {}
-    Object.keys(propertyGroup).forEach(function (k) {
-        obj[k] = propertyGroup[k]
-    })
-    if (isValidPropertyFilter(obj)) {
-        flattenedProperties.push(obj)
+export function convertCohortCriteriaToCohortCriteriaGroup(
```

Comment:
One of the values of types is it lessens the need for verbose names such as this, eg `convert`.

```
import { convert as convertCriteria } from './criteriaUtils';
import { convert as convertFoobar } from './foobarUtils';

const grouping = convertCriteria(cohortCriteria);
```

This would require most of these criteria functions to live in their own file, to prevent name collisions. (Which is probably an improvement from the above comment)